### DISCUSSION TITLE
Is Rust package fully working?

### DISCUSSION THREAD

**Discussion by bubuntux** [AstroNvim/astrocommunity#674] [Q&A]
I just installed astronvim with the rust community package and doesn't seems fully functional, 
it is not able to display documentation of built-in methods such as trim or chars for a string, also the autocomplete doesn't suggest them i think there is something missing on the LSP.

setup
```
docker run -w /root -it --rm alpine:edge sh -uelic '
  apk add bash git lua nodejs npm lazygit bottom python3 go neovim ripgrep alpine-sdk rust cargo rustfmt --update
  git clone --depth 1 https://github.com/AstroNvim/AstroNvim ~/.config/nvim
  mkdir -p ~/.config/nvim/lua/user/plugins/
  echo "return { \"AstroNvim/astrocommunity\", { import = \"astrocommunity.pack.rust\" }, }" >> ~/.config/nvim/lua/user/plugins/community.lua
  cargo init abc
  nvim abc/src/main.rs
'
``` 
after it finish to load and install all the dependencies the LSP wont start automatically since it just got installed so we need to manually start it using 
```
:LspStart rust-analyzer 
```
after that a lot of things work out-of-the-box like syntax, formatting, errors/warnings and such, but if you type `"ABC".` it won't suggest methods such as trim or chars and when manually typed them and then try to get the documentation about them using `K` nothing happens.

Am i missing something obvious?

**Comment by M-Komorek** [comment]
Same problem with Rust package. In addiction, the same behavior occurs for example for cpp pack.
Any idea how to solve it? Are we missing something obvious?

**Comment by river-studio-main** [comment]
a few years forward and still not seeming to work out of the box, but then i noticed that when opening a rust file it tells me that rust analyzer crushed. weird, but it explains all the issues of not having completion and etc. out of the box, right? so i check and apparently while using nightly builds on macos with apple silicon chip rustup does not install the correct rust analyzer for the toolchain and it has to be installed manually for everything to start clicking: `rustup component add rust-analyzer --toolchain nightly`. and then everything worked as expected out of the box, i didn't even have to restart nvim or the plugins.
i hope this saves someone a few hours of fighting with configs and toaster messages :3

**Reply by ALameLlama** [reply]
The [Rust Pack](https://astronvim.github.io/astrocommunity/#rust) does have a note saying you'll need to add the rust-analyzer  via rustup
